node/overseer/overseer-gen/proc-macro/src/impl_builder.rs

```diff
@@ -85,11 +50,15 @@ pub(crate) fn impl_builder(info: &OverseerInfo) -> proc_macro2::TokenStream {
 	let channel_name_rx = &info.channel_names_without_wip("_rx");
 	let channel_name_unbounded_rx = &info.channel_names_without_wip("_unbounded_rx");
 
-	let baggage_generic_ty = &info.baggage_generic_types();
 	let baggage_name = &info.baggage_names();
-	let baggage_ty = &info.baggage_types();
-
-	let subsystem_ctx_name = format_ident!("{}SubsystemContext", overseer_name);
+	let baggage_generic_ty = &info.baggage_generic_types();
+	let baggage_placeholder_ty = baggage_name
```

Comment:
nit: I think palceholder is slight misnomer, it's really the init state as a generic iiuc, no idea to express this concisely